### DISCUSSION TITLE
How to Limit Custom Field Values Pulled

### DISCUSSION THREAD

**Discussion by tegaidev** [pnp/pnpcore#1244] [Q&A]
New user and I can't figure this one out after a full day of trying every suggestion and syntax I could find. Using version 1.8 (can't use latest). 

No matter what I do, when I retrieve list items from my list, I get every single field value. I need to pull thousands of items from a list with >50 custom columns, but I only want the data from one or two of those columns.

This is basically what I'm trying right now:

```
var myList = await context.Web.Lists.GetByIdAsync(new Guid('list-guid-here'), p => p.Title);

var items = await myList.Items.QueryProperties(p => p["MyCustomFieldInternalName"]).ToListAsync();
```

What am I missing here? I don't want to have to wait for all that unnecessary data every time.

Thanks you!

EDIT:  In Fiddler I can see that the $select element is missing from the REST call. I expected that calling QueryProperties and specifying the field internal name would be the equivalent, but I am misunderstanding?

EDIT2: Apparently QueryProperties has no effect on the Items collection? I want to limit what is returned in the Values collection for each item. HOW?  If I'm not allowed, why? Doesn't make sense to me on the surface.

EDIT3:  Starting to look like there just is no such ability built into PnP.Core methods, except to build the request and execute it on the context yourself with the necessary $select element with ExecuteRequestAsync. I can't imagine that it might be sending 100x more data than necessary with each item would not be an issue brought up before, so I have to be doing something wrong here - this seems like missing basic functionality.

**Comment by ajcaterino** [comment]
This is just a thought and I have not tried any of this, but QueryProperties returns IQueryable. I'm assuming it would work similar to EF where you can do:

`var items = await myList.Items.QueryProperties(p => p["MyCustomFieldInternalName"]).Where(p => p.SomeProperty == "Something").ToListAsync();`

You may have to do this to load the Items into the list object:

`var items = await myList.Load(p => p.Items.QueryProperties(p => p["MyCustomFieldInternalName"]).Where(p => p.SomeProperty == "Something"));`

**Reply by jansenbe** [reply]
@tegaidev : I'd recommend this approach for large scale querying: https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#using-paging-with-listdataasstream
